presto-main/src/main/java/io/prestosql/operator/scalar/JsonOperators.java

```diff
@@ -340,7 +340,12 @@ public static Slice castFromTimestamp(ConnectorSession session, @SqlType(TIMESTA
         try {
             SliceOutput output = new DynamicSliceOutput(25);
             try (JsonGenerator jsonGenerator = createJsonGenerator(JSON_FACTORY, output)) {
-                jsonGenerator.writeString(printTimestampWithoutTimeZone(session.getTimeZoneKey(), value));
+                if (session.isLegacyTimestamp()) {
+                    jsonGenerator.writeString(printTimestampWithoutTimeZone(session.getTimeZoneKey(), value));
+                }
+                else {
+                    jsonGenerator.writeString(printTimestampWithoutTimeZone(value));
```

Comment:
> Fix legacy/new timestamp cast to json

This should probably be "Fix timestamp CAST to JSON in new semantics" or something, as you are not fixing anything in the legacy semantics (right?)

Reply:
Yes